### PR TITLE
added tests for location

### PR DESCRIPTION
**Pull request by serainaschraff**
please check if this is alright :)

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 1]
```diff
@@ -913,11 +913,11 @@ public void updateLocation(long userId, double latitude, double longitude) {
...
     // else keep them at default 0, i.e. do nothing
-    if (latitude != 0 && longitude != 0) {
-      user.setLatitude(latitude);
-      user.setLongitude(longitude);
+    if (latitude == 0 && longitude == 0) {
+        return;
     }
-
+    user.setLatitude(latitude);
+    user.setLongitude(longitude);
```
Comment (daveking7) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:920]: Why did you change that though?

File: src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java [hunk 2]
```diff
@@ -611,6 +612,20 @@ void getLoginStatus_success() throws Exception {
         //TODO: read the returned value
     }
 
+    @Test
+    void updateLocation_sucess() throws Exception {
+        User user1 = fillUser(1L, "1");
+        doNothing().when(userService).checkSpecificAccess(user1.getToken(), user1.getId());
+
+        LocationPostDTO locationDTO = new LocationPostDTO();
```
Comment (daveking7) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java:620]: Shouldn't it be now called LoacationPutDTO (@TwoDigitsOneNumber ) ?

**Review by daveking7 — APPROVED** [review]
Looks good

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java (hunk 1 above)
Comment (serainaschraff) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:920]: Because you could be on the equator or the greenwich line and then you would want it to update to that location. 0,0 is highly unlikely that it is ever reached so this is used to differenciate between a valid and invalid change of the location.

Comment (daveking7) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:920]: Ah yes, stupid me. But you could just replaced && with || ^^
